daemon/run.go

```diff
@@ -0,0 +1,319 @@
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+
+	"github.com/akitasoftware/akita-cli/apispec"
+	"github.com/akitasoftware/akita-cli/har_loader"
+	"github.com/akitasoftware/akita-cli/plugin"
+	"github.com/akitasoftware/akita-cli/rest"
+	"github.com/akitasoftware/akita-cli/trace"
+	"github.com/akitasoftware/akita-cli/util"
+	"github.com/akitasoftware/akita-libs/akid"
+	kgxapi "github.com/akitasoftware/akita-libs/api_schema"
+	"github.com/akitasoftware/akita-libs/sampled_err"
+)
+
+const WITNESS_BUFFER_SIZE = 10_000
+
+type Witness = har_loader.CustomHAREntry
+
+type Args struct {
+	// Required args.
+	ClientID akid.ClientID
+	Domain   string
+
+	// Optional args.
+	PortNumber uint16
+
+	Plugins []plugin.AkitaPlugin
+}
+
+// A wrapper around HTTP request handlers for handling HTTP responses generically.
+type httpHandler func(*http.Request) HTTPResponse
+
+var cmdArgs Args
+
+func Run(args Args) error {
+	cmdArgs = args
+	router := mux.NewRouter().StrictSlash(true)
+
+	// Register an endpoint for creating a new learning session.
+	router.Handle("/services/{serviceName}/learn/sessions", httpHandler(createLearnSession)).Methods("POST")
```

Comment:
I am confused about how the pieces plug together here; how does mux know what to do with the HTTPResponse type that you defined?  Or the ClientResponse interface, which you also defined?

It's not a big deal, but if it works by satisfying some existing interface, I have found the idiom

var _ ClientResponse = &HttpResponse{}

to be useful for documenting the type assumption (whatever it is?) and giving the compiler a place to check the type assertion even if there's an interface{} or other non-type-checkable call involved.

Reply:
Addressed in 77443df.

Reply:
Got it, thanks, didn't see the ServeHTTP function.